advanced-achievements-plugin/src/main/java/com/hm/achievement/command/executable/AbstractParsableCommand.java

```diff
@@ -43,16 +50,23 @@ public void extractConfigurationParameters() {
 	@Override
 	void onExecute(CommandSender sender, String[] args) {
 		String searchedName = args[args.length - 1];
-		Player player = Bukkit.getOnlinePlayers().stream().filter(p -> p.getName().equalsIgnoreCase(searchedName))
-				.findFirst().orElse(null);
-
-		// If player not found or is offline.
-		if (player == null) {
-			sender.sendMessage(pluginHeader + StringUtils.replaceOnce(langPlayerOffline, "PLAYER", searchedName));
+		Entity[] entities = CommandUtils.getTargets(sender, searchedName);
+		if (entities == null) {
+			sender.sendMessage(pluginHeader + langEntityNotPlayer);
 			return;
 		}
+		for (int i = 0; i < entities.length; i++) {
+			if (entities[i] == null) {
+				sender.sendMessage(pluginHeader + StringUtils.replaceOnce(langPlayerOffline, "PLAYER", searchedName));
+				break;
+			}
+			try {
+				onExecuteForPlayer(sender, args, (Player) entities[i]);
+			} catch (ClassCastException e) {
```

Comment:
Rather than using the exception for control flow, I suggest using a condition guard:
````
if (entities[i] instanceof Player) {
	onExecuteForPlayer(sender, args, (Player) entities[i]);
} else {
	sender.sendMessage(pluginHeader + langEntityNotPlayer);
}
````